### DISCUSSION TITLE
Delete markers and non-versioned buckets

### DISCUSSION THREAD

**Discussion by nicolamori** [minio/minio#20634] [Q&A]
I'd need some help with these topics since I think I don't understand something fundamental. I have a non-versioned (I suppose) bucket:
```
$ mcli version info herd-storage/gitlab-backup/                   
herd-storage/gitlab-backup/ versioning is suspended
```
Now, if i put a file inside it:
```
$ mcli cp docker-compose.yaml herd-storage/gitlab-backup                                               
...ysis/HERD/docker-compose/docker-compose.yaml: 8.99 KiB / 8.99 KiB
$ mcli ls herd-storage/gitlab-backup/docker-compose.yaml
[2024-11-11 12:56:11 CET] 9.0KiB STANDARD docker-compose.yaml
```
and then I delete it then a delete marker is created:
```
mcli rm herd-storage/gitlab-backup/docker-compose.yaml
Created delete marker `herd-storage/gitlab-backup/docker-compose.yaml`.
```
and the object is actually still present in a DEL state:
```
$ mcli ls --versions herd-storage/gitlab-backup/docker-compose.yaml
[2024-11-11 12:57:04 CET]     0B STANDARD null v1 DEL docker-compose.yaml
```

From what I understood from the web documentation, delete markers are created only for versioned buckets; so either I misunderstood, or my bucket is versioned and I'm misinterpreting the output of `mcli version info`.

Can anybody please help me understanding what I'm doing wrong?

**Comment by nicolamori** [comment]
Looking at the versioning state of all my buckets I see that some are marked with `versioning is suspended` and others as `un-versioned`. For the latters `mcli rm` actually removes the file without creating a delete marker. So I think that converting my bucket to unversioned might fix my problem, but  on the web I can't find how to do so (`mcli version` seems able only to suspend versioning and not disabling it), so I even doubt it is possible to disable versioning once it has been enabled.

**Comment by kuznetsov-as** [comment]
@nicolamori Were you able to solve this problem with delete markers? I also don't see how I can clear my bucket of them now.

**Reply by nicolamori** [reply]
I had to create a new bucket with a temporary name, move all data in it, delete the old one, create a new bucket with the same name of the old one, move in it all data from temporary bucket, and finally delete the latter. It could have been less cumbersome if at least bucket renaming was possible...
